src/main/scala/org/fortysevendeg/sparksftp/common/HiveUserData.scala

```diff
@@ -39,9 +44,29 @@ object HiveUserData {
     salariesDataFromHive.show(false)
 
     //Excerpt from the joined table
-    sparkSession.sql("select name,salary from user_salary").show(false)
+    val userSalaries = sparkSession.sql("select name,salary from user_salary")
+    userSalaries.show(false)
+
+    HiveUserData(userDataFromHive, salariesDataFromHive, userSalaries)
+  }
+
+  def persistAndReadUserData(
+      sparkSession: SparkSession,
+      users: DataFrame,
+      salaries: DataFrame
+  ): IO[HiveUserData] = IO {
+    HiveUserData.persistUserData(sparkSession, users, salaries)
+    HiveUserData.readUserData(sparkSession)
+  }
 
-    (userDataFromHive, salariesDataFromHive)
+  def calculateAndPersistNewSalary(
+      sparkSession: SparkSession,
+      userSalaries: DataFrame
+  ): IO[Dataset[Row]] = IO {
+    val newSalaries =
+      userSalaries.withColumn("new_salary", (userSalaries("salary") * 1.1).cast(IntegerType))
```

Comment:
Why are we casting to Integer?

Reply:
Basically, the mode of operating with columns, they come with no basic type.